### ISSUE TITLE
Upserts

### ISSUE THREAD

**Issue by neverfox** [SparkFund/spec-tacular#26]
I'm just getting familiar with spec-tacular's API but it doesn't currently seem to take advantage of Datomic's lookup ref behavior to do assertions (essentially upserts). That capability is something we use (abuse?) a lot when streaming new data into the database. This is nice because you don't have to make go through the steps of first fetching the entities you want to create or update, and then taking different paths depending on the reality of the situation. However, that seems to be the way you have to operate with spec-tacular if only working with the `create!` and `assoc!` functions (given that the former would throw an exception and the latter requires a "an object representation of an entity on the database"). Is there currently a way to get this kind of behavior from spec-tacular or would that require the addition of something like `upsert!`? Is that something have considered or would consider?

**Comment by calvis** [comment]
Yeah, I've thought about this before but haven't nailed down an API.  I enjoy the current behavior of `create!` as a function you can use when you think the object shouldn't already be on the database, but I think `upsert!` could be achieved without too much difficulty.  I'd think spec-tacular could do it mostly implicitly, too, since unique fields are defined in the spec; all it would have to do is substitute lookup refs for temp ids wherever possible?

**Comment by neverfox** [comment]
Yeah, that's what I was thinking too. It would just have to fall back to what it already knows + Datomic's inherent behavior.
